dds/DCPS/QOS_XML_Handler/QOS_XML_Loader.cpp

```diff
@@ -8,76 +8,76 @@ OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
 namespace OpenDDS {
 namespace DCPS {
 
-  QOS_XML_Loader::QOS_XML_Loader (void)
+  QOS_XML_Loader::QOS_XML_Loader(void)
   {
   }
 
-  QOS_XML_Loader::~QOS_XML_Loader (void)
+  QOS_XML_Loader::~QOS_XML_Loader(void)
   {
   }
 
 
-  char*
-  QOS_XML_Loader::get_xml_file_name (const char * qos_profile)
+  ACE_TCHAR*
+  QOS_XML_Loader::get_xml_file_name(const ACE_TCHAR * qos_profile)
   {
     if (qos_profile)
       {
-        char* buf = ACE_OS::strdup (qos_profile);
-        ACE_Tokenizer_T<char> tok (buf);
-        tok.delimiter_replace ('#', 0);
-        const char * file_name = tok.next ();
+        ACE_TCHAR* buf = ACE_OS::strdup(qos_profile);
+        ACE_Tokenizer_T<ACE_TCHAR> tok(buf);
+        tok.delimiter_replace('#', 0);
+        const ACE_TCHAR * file_name = tok.next();
 
         if (file_name == 0)
           {
-            ACE_OS::free (buf);
+            ACE_OS::free(buf);
             if (DCPS_debug_level > 5)
               {
-                ACE_ERROR ((LM_ERROR,
-                              "get_xml_file_name <%C> - "
-                              "Error: malformed qos_profile. Expected format: "
-                              "<xml_file_base_name>#<profile_name>\n",
-                              qos_profile));
+                ACE_ERROR((LM_ERROR,
+                             "get_xml_file_name <%s> - "
+                             "Error: malformed qos_profile. Expected format: "
+                             "<xml_file_base_name>#<profile_name>\n",
+                             qos_profile));
               }
             return 0;
           }
 
-        char * ret = (char*)ACE_OS::malloc (ACE_OS::strlen (file_name) + 5);
-        ret = ACE_OS::strcpy (ret, file_name);
-        ret = ACE_OS::strcat (ret, ".xml");
-        ACE_OS::free (buf);
+        ACE_TCHAR * ret = (ACE_TCHAR*) ACE_OS::malloc(ACE_OS::strlen(file_name) + 5);
```

Comment:
This doesn't look right, strlen() measures characters while malloc() wants bytes.  In general there's no reason to avoid using a class like TString or even std::basic_string that does all of this automatically.  (I realize it was like this before.)  Please see the WCHAR notes (in docs/design) for more info on using switchable character types.